Typescript Build Error on Accepts

### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised


### Issue

Hey there, 

Not sure whether this should be raised as a bug or not but when trying to install this in a TS project as this may be intended but I received the following error:

https://github.com/seek-oss/slowify/actions/runs/5167692569/jobs/9308739911

```
Error: tsc      │ node_modules/@fastify/accepts/types/index.d.ts(1,25): error TS7016: Could not find a declaration file for module 'accepts'. '/home/runner/work/slowify/slowify/node_modules/accepts/index.js' implicitly has an 'any' type.
```

Could `@types/accepts` be made a dependency for this library to avoid consumers needing to install it themselves?

--- a/package.json
+++ b/package.json
@@ -6,7 +6,6 @@
   "types": "types/index.d.ts",
   "devDependencies": {
     "@fastify/pre-commit": "^2.0.2",
-    "@types/accepts": "^1.3.5",
     "@types/node": "^20.1.0",
     "fastify": "^4.0.0-rc.2",
     "request": "^2.88.0",
@@ -16,6 +15,7 @@
     "tsd": "^0.28.0"
   },
   "dependencies": {
+    "@types/accepts": "^1.3.5",
     "accepts": "^1.3.5",
     "fastify-plugin": "^4.0.0"
   },

--- a/package.json
+++ b/package.json
@@ -6,7 +6,6 @@
   "types": "types/index.d.ts",
   "devDependencies": {
     "@fastify/pre-commit": "^2.0.2",
-    "@types/accepts": "^1.3.5",
     "@types/node": "^20.1.0",
     "fastify": "^4.0.0-rc.2",
     "request": "^2.88.0",

--- a/package.json
+++ b/package.json
@@ -6,6 +6,7 @@
   "types": "types/index.d.ts",
   "devDependencies": {
     "@fastify/pre-commit": "^2.0.2",
+    "@types/accepts": "^1.3.5",
     "@types/node": "^20.1.0",
     "fastify": "^4.0.0-rc.2",
     "request": "^2.88.0",
@@ -15,7 +16,6 @@
     "tsd": "^0.28.0"
   },
   "dependencies": {
-    "@types/accepts": "^1.3.5",
     "accepts": "^1.3.5",
     "fastify-plugin": "^4.0.0"
   },

--- a/types/index.d.ts
+++ b/types/index.d.ts
@@ -2,6 +2,8 @@ import { FastifyPluginCallback } from "fastify"
 
 declare module "fastify" {
   // Copied from https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/accepts/index.d.ts
+  // Definitions by: Stefan Reichel <https://github.com/bomret>
+  //                 Brice BERNARD  <https://github.com/brikou>
   interface Accepts {
     /**
      * Return the first accepted charset. If nothing in `charsets` is accepted, then `false` is returned.

--- a/types/index.d.ts
+++ b/types/index.d.ts
@@ -1,7 +1,92 @@
-import { Accepts } from "accepts"
 import { FastifyPluginCallback } from "fastify"
 
 declare module "fastify" {
+  // Copied from https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/accepts/index.d.ts
+  interface Accepts {
+    /**
+     * Return the first accepted charset. If nothing in `charsets` is accepted, then `false` is returned.
+     * If no charsets are supplied, all accepted charsets are returned, in the order of the client's preference
+     * (most preferred first).
+     */
+    charset(): string[];
+    charset(charsets: string[]): string | false;
+    charset(...charsets: string[]): string | false;
+  
+    /**
+     * Return the first accepted charset. If nothing in `charsets` is accepted, then `false` is returned.
+     * If no charsets are supplied, all accepted charsets are returned, in the order of the client's preference
+     * (most preferred first).
+     */
+    charsets(): string[];
+    charsets(charsets: string[]): string | false;
+    charsets(...charsets: string[]): string | false;
+  
+    /**
+     * Return the first accepted encoding. If nothing in `encodings` is accepted, then `false` is returned.
+     * If no encodings are supplied, all accepted encodings are returned, in the order of the client's preference
+     * (most preferred first).
+     */
+    encoding(): string[];
+    encoding(encodings: string[]): string | false;
+    encoding(...encodings: string[]): string | false;
+  
+    /**
+     * Return the first accepted encoding. If nothing in `encodings` is accepted, then `false` is returned.
+     * If no encodings are supplied, all accepted encodings are returned, in the order of the client's preference
+     * (most preferred first).
+     */
+    encodings(): string[];
+    encodings(encodings: string[]): string | false;
+    encodings(...encodings: string[]): string | false;
+  
+    /**
+     * Return the first accepted language. If nothing in `languages` is accepted, then `false` is returned.
+     * If no languaes are supplied, all accepted languages are returned, in the order of the client's preference
+     * (most preferred first).
+     */
+    language(): string[];
+    language(languages: string[]): string | false;
+    language(...languages: string[]): string | false;
+  
+    /**
+     * Return the first accepted language. If nothing in `languages` is accepted, then `false` is returned.
+     * If no languaes are supplied, all accepted languages are returned, in the order of the client's preference
+     * (most preferred first).
+     */
+    languages(): string[];
+    languages(languages: string[]): string | false;
+    languages(...languages: string[]): string | false;
+  
+    /**
+     * Return the first accepted language. If nothing in `languages` is accepted, then `false` is returned.
+     * If no languaes are supplied, all accepted languages are returned, in the order of the client's preference
+     * (most preferred first).
+     */
+    lang(): string[];
+    lang(languages: string[]): string | false;
+    lang(...languages: string[]): string | false;
+  
+    /**
+     * Return the first accepted language. If nothing in `languages` is accepted, then `false` is returned.
+     * If no languaes are supplied, all accepted languages are returned, in the order of the client's preference
+     * (most preferred first).
+     */
+    langs(): string[];
+    langs(languages: string[]): string | false;
+    langs(...languages: string[]): string | false;
+  
+    /**
+     * Return the first accepted type (and it is returned as the same text as what appears in the `types` array). If nothing in `types` is accepted, then `false` is returned.
+     * If no types are supplied, return the entire set of acceptable types.
+     *
+     * The `types` array can contain full MIME types or file extensions. Any value that is not a full MIME types is passed to `require('mime-types').lookup`.
+     */
+    type(types: string[]): string[] | string | false;
+    type(...types: string[]): string[] | string | false;
+    types(types: string[]): string[] | string | false;
+    types(...types: string[]): string[] | string | false;
+  }
+
   interface FastifyRequest extends Accepts {
     accepts(): Accepts
   }

--- a/types/index.test-d.ts
+++ b/types/index.test-d.ts
@@ -6,7 +6,27 @@ const app = fastify()
 app.register(accepts)
 
 app.get("/", (request, reply) => {
-  request.accepts()
+  const accept = request.accepts();
+
+  const charsets = accept.charsets();
+  const charsetFromEmpty = accept.charset();
+  const charsetFromParams = accept.charset("json", "text");
+  const charsetFromArray = accept.charset(["json", "text"]);
+
+  const encodings = accept.encodings();
+  const encodingFromEmpty = accept.encoding();
+  const encodingFromParams = accept.encoding("json", "text");
+  const encodingFromArray = accept.encoding(["json", "text"]);
+
+  const languages = accept.languages();
+  const languageFromEmpty = accept.language();
+  const languageFromParams = accept.language("json", "text");
+  const languageFromArray = accept.language(["json", "text"]);
+
+  const types = accept.types();
+  const typeFromEmpty = accept.type();
+  const typeFromParams = accept.type("json", "text");
+  const typeFromArray = accept.type(["json", "text"]);
 
   request.charset()
   request.charsets()
@@ -66,3 +86,4 @@ app.get("/", (request, reply) => {
 
   reply.send({ hello: "world" })
 })
+

